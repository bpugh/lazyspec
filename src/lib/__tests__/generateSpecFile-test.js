const fs = require('fs');

const generateSpecFile = require('../generateSpecFile');

function expectSpec(moduleName) {
  const unitPath = `spec/fixtures/${moduleName}.js`;
  const specPath = `spec/fixtures/__tests__/${moduleName}-test.js`;
  const src = fs.readFileSync(unitPath).toString();

  const spec = generateSpecFile(moduleName, specPath, unitPath, src);
  expect(spec).toMatchSnapshot();
}

describe('generateSpecFile', () => {
  it('should describe simple exports', () => {
    expectSpec('esExports');
  });

  it('should describe an ES5 component', () => {
    expectSpec('CreateClass');
  });

  it('should describe an ES6 component', () => {
    expectSpec('SafeComponent');
  });

  it('should describe an unsafe ES6 component', () => {
    expectSpec('UnsafeComponent');
  });

  it('should describe an ES6 with Flow component', () => {
    expectSpec('FlowComponent');
  });

  it('should describe a stateless function component', () => {
    expectSpec('StatelessFunction');
  });

  it('should describe multi-exports', () => {
    expectSpec('MultipleComponents');
  });
});
