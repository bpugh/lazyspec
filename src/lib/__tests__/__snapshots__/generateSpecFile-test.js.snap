exports[`generateSpecFile should describe a stateless function component 1`] = `
"/* @lazyspec (remove to manage manually) */
/* eslint-disable */
import StatelessFunction from \'../StatelessFunction.js\';

import React from \'react\';
import { shallow } from \'enzyme\';
import { shallowToJson } from \'enzyme-to-json\';

describe(\'StatelessFunction\', () => {
  it(\'exists\', () => {
    expect(StatelessFunction).toBeTruthy();
  });

  it(\'renders\', () => {
    const comp = <StatelessFunction  />;
    const wrapper = shallow(comp);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
"
`;

exports[`generateSpecFile should describe an ES5 component 1`] = `
"/* @lazyspec (remove to manage manually) */
/* eslint-disable */
import CreateClass from \'../CreateClass.js\';

import React from \'react\';
import { shallow } from \'enzyme\';
import { shallowToJson } from \'enzyme-to-json\';

describe(\'CreateClass\', () => {
  it(\'exists\', () => {
    expect(CreateClass).toBeTruthy();
  });

  it(\'renders\', () => {
    const comp = <CreateClass  />;
    const wrapper = shallow(comp);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
"
`;

exports[`generateSpecFile should describe an ES6 component 1`] = `
"/* @lazyspec (remove to manage manually) */
/* eslint-disable */
import SafeComponent from \'../SafeComponent.js\';

import React from \'react\';
import { shallow } from \'enzyme\';
import { shallowToJson } from \'enzyme-to-json\';

describe(\'SafeComponent\', () => {
  it(\'exists\', () => {
    expect(SafeComponent).toBeTruthy();
  });

  it(\'renders\', () => {
    const comp = <SafeComponent children=\"mockElement\" />;
    const wrapper = shallow(comp);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
"
`;

exports[`generateSpecFile should describe an ES6 with Flow component 1`] = `
"/* @lazyspec (remove to manage manually) */
/* eslint-disable */
import FlowComponent from \'../FlowComponent.js\';

import React from \'react\';
import { shallow } from \'enzyme\';
import { shallowToJson } from \'enzyme-to-json\';

describe(\'FlowComponent\', () => {
  it(\'exists\', () => {
    expect(FlowComponent).toBeTruthy();
  });

  it(\'renders\', () => {
    const comp = <FlowComponent label=\"mockString\" children=\"mockAny\" />;
    const wrapper = shallow(comp);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
"
`;

exports[`generateSpecFile should describe an unsafe ES6 component 1`] = `
"/* @lazyspec (remove to manage manually) */
/* eslint-disable */
import UnsafeComponent from \'../UnsafeComponent.js\';

import React from \'react\';
import { shallow } from \'enzyme\';
import { shallowToJson } from \'enzyme-to-json\';

describe(\'UnsafeComponent\', () => {
  it(\'exists\', () => {
    expect(UnsafeComponent).toBeTruthy();
  });

  xit(\'renders\', () => {
    const comp = <UnsafeComponent foo=\"TODO mock\" />;
    const wrapper = shallow(comp);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
"
`;

exports[`generateSpecFile should describe multi-exports 1`] = `
"/* @lazyspec (remove to manage manually) */
/* eslint-disable */
import MultipleComponents, {Bold} from \'../MultipleComponents.js\';

describe(\'MultipleComponents\', () => {
  it(\'exists\', () => {
    expect(MultipleComponents).toBeTruthy();
  });
});
describe(\'Bold\', () => {
  it(\'exists\', () => {
    expect(Bold).toBeTruthy();
  });
});
"
`;

exports[`generateSpecFile should describe simple exports 1`] = `
"/* @lazyspec (remove to manage manually) */
/* eslint-disable */
import esExports, {refConst, refFunc, inlineConst, inlineFunc} from \'../esExports.js\';

describe(\'esExports\', () => {
  it(\'exists\', () => {
    expect(esExports).toBeTruthy();
  });
});
describe(\'refConst\', () => {
  it(\'exists\', () => {
    expect(refConst).toBeTruthy();
  });
});
describe(\'refFunc\', () => {
  it(\'exists\', () => {
    expect(refFunc).toBeTruthy();
  });
});
describe(\'inlineConst\', () => {
  it(\'exists\', () => {
    expect(inlineConst).toBeTruthy();
  });
});
describe(\'inlineFunc\', () => {
  it(\'exists\', () => {
    expect(inlineFunc).toBeTruthy();
  });
});
"
`;
